src/main/java/gregtech/api/util/GTUtility.java

```diff
@@ -1024,9 +1038,124 @@ public static int combineRGB(@Range(from = 0, to = 255) int r, @Range(from = 0,
+        if (euItem != null) {
+            return (float) euItem.getCharge() / euItem.getMaxCharge();
+        }
+
+        IEnergyStorage rfItem = stack.getCapability(CapabilityEnergy.ENERGY, null);
+        if (rfItem != null) {
+            return (float) rfItem.getEnergyStored() / rfItem.getMaxEnergyStored();
+        }
+
+        return -1.0f;
+    }
+
+    /**
+     * Lerp between two ARGB colors
+     *
+     * @param start    the start point of the lerp
+     * @param end      the end point of the lerp
+     * @param position the position of the lerp. Must be between 0 and 1.
+     * @return the lerped color value
+     */
+    public static int argbLerp(int start, int end, float position) {
+        int aStart = (start >> 24 & 0xFF);
+        int rStart = (start >> 16 & 0xFF);
+        int gStart = (start >> 8 & 0xFF);
+        int bStart = (start & 0xFF);
+
+        int aEnd = (end >> 24 & 0xFF);
+        int rEnd = (end >> 16 & 0xFF);
+        int gEnd = (end >> 8 & 0xFF);
+        int bEnd = (end & 0xFF);
+
+        int aFinal = (int) (aStart + (aEnd - aStart) * position);
+        int rFinal = (int) (rStart + (rEnd - rStart) * position);
+        int gFinal = (int) (gStart + (gEnd - gStart) * position);
+        int bFinal = (int) (bStart + (bEnd - bStart) * position);
+
+        return (aFinal << 24) | (rFinal << 16) | (gFinal << 8) | bFinal;
+    }
+
+    /**
+     * Swaps int[a] with int[b].
+     */
+    public static void swap(int[] swappingArray, int a, int b) {
+        int tmp = swappingArray[a];
+        swappingArray[a] = swappingArray[b];
+        swappingArray[b] = tmp;
+    }
+
+    public static void shuffle(int[] shufflingArray) {
+        for (int index = shufflingArray.length; index > 1; index--) {
+            swap(shufflingArray, index - 1, GTValues.RNG.nextInt(index));
+        }
+    }
+
     public static boolean areFluidStacksEqual(@Nullable FluidStack a, @Nullable FluidStack b) {
         if (a == b) return true;
         if (a == null) return false;
         return a.isFluidEqual(b);
     }
+
+    /**
+     * Check if the provided month and day is today with a given time zone.
+     */
+    public static boolean isToday(@NotNull MonthDay monthDay, @Nullable ZoneId tz) {
+        MonthDay now = tz == null ? MonthDay.now() : MonthDay.now(tz);
+        return now.equals(monthDay);
+    }
+
+    /**
+     * Check if the provided month and day is today.
+     */
+    public static boolean isToday(@NotNull MonthDay monthDay) {
+        return isToday(monthDay, null);
+    }
+
+    /**
+     * Check if today is April 1st.
+     */
+    public static boolean isAprilFools() {
+        return isToday(GTValues.APRIL_FOOLS);
+    }
+
+    /**
+     * Check if today is the day of, or eve of Xmas.
+     */
+    public static boolean isXMAS() {
+        return isToday(GTValues.XMAS_EVE) || isToday(GTValues.XMAS);
+    }
```

Comment:
These two should check the special events config (Probably the easiest place to do it would be here), as none of the new implementations do so locally. Previously it was in the supplier construction.